extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,461 @@
+                    default:
+                        return null;
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public String target(final HttpRequest request) {
+            return request.uri();
+        }
+
+        @Override
+        public String route(final HttpRequest request) {
+            return request.uri().length() > 1 ? request.uri() : null;
+        }
+
+        @Override
+        public String scheme(final HttpRequest request) {
+            if (request instanceof HttpServerRequest) {
+                return ((HttpServerRequest) request).scheme();
+            }
+            return null;
+        }
+
+        @Override
+        public String serverName(final HttpRequest request, final HttpResponse response) {
+            return request.remoteAddress().hostName();
+        }
+
+        @Override
+        public String method(final HttpRequest request) {
+            return request.method().name();
+        }
+
+        @Override
+        public List<String> requestHeader(final HttpRequest request, final String name) {
+            return request.headers().getAll(name);
+        }
+
+        @Override
+        public Long requestContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long requestContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Integer statusCode(final HttpRequest request, final HttpResponse response) {
+            return response.statusCode();
+        }
+
+        @Override
+        public Long responseContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long responseContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public List<String> responseHeader(final HttpRequest request, final HttpResponse response, final String name) {
+            return response.headers().getAll(name);
+        }
+    }
+
+    // TODO - Ideally this should use HttpSpanNameExtractor, but to keep the name without the slash we use our own.
+    private static class ServerSpanNameExtractor implements SpanNameExtractor<HttpRequest> {
+        private final HttpServerAttributesGetter<HttpRequest, HttpResponse> serverAttributesExtractor;
+
+        ServerSpanNameExtractor(
+                final HttpServerAttributesGetter<HttpRequest, HttpResponse> serverAttributesExtractor) {
+            this.serverAttributesExtractor = serverAttributesExtractor;
+        }
+
+        @Override
+        public String extract(final HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpServerRequest) {
+                String path = URI.create(httpRequest.uri()).getPath();
```

Comment:
I'm always worried about encoding (space and such kinds of things). I do not believe you can get an encoding issue on the path, but better check.